### PR TITLE
Telia play history sync

### LINKED ISSUE #40: Request: Telia Play
Hi! I just found your project while searching for a way to scrobble or sync Telia Play to trakt.tv. 

I tried following your development guide but my knowledge of chrome plugin development is non existent and my time is quite limited (parent with small children). 

However it seems like Telia has a closed API with many endpoints when inspecting the network requests. 

For example this call which fetches all "Continue watching" items:
https://ottapi.prod.telia.net/web/se/continuewatching/rest/secure/v2/watchedItems/list/?fromIndex=100&toIndex=199&protocols=dash&resolutions=sd&serviceTypes=SVOD,VOD

Which returns data like:
```
...
		{
			"mediaId": "1963970510",
			"assetId": "18251879",
			"duration": 2520,
			"position": 0,
			"timestamp": 1601842508186,
			"hidden": false,
			"serviceType": "SVOD",
			"mediaObjectType": "OneCoverObject",
			"mediaObject": {
				"validTo": "2023-04-12T21:59:59Z",
				"validFrom": "2020-04-13T05:00:00Z",
				"type": "VIDEO",
				"totalAssets": 6,
				"title": "03:01 Slowly Slowly Catchy Monkey - Killing Eve",
				"subtitleLanguages": [],
				"storeType": "SVOD",
				"storeName": "HBO Nordic",
				"storeIds": [
					"8925",
					"8924",
					"8959"
				],
				"storeId": "8925",
				"seriesTitle": "Killing Eve",
				"seriesId": "1489517736",
				"seasonId": "1489517736_03",
				"seasonNumber": 3,
				"renttime": 10800000,
				"productionYear": "2020",
				"portraitImages": [
					"/HBOFullHD/5803798.jpg",
					"/HBOHD/5803799.jpg",
					"/HBO/5803800.jpg"
				],
				"parentalcontrol": 0,
				"loopId": "1963970510",
				"logoUrl": "/images/logos/hbo_logo_331x38.png",
				"length": 42,
				"language": "swe",
				"landscapeImages": [],
				"episodeNumber": 1,
				"directors": [
					"Waller-Bridge,Phoebe"
				],
				"contentproviderName": "HBO",
				"contentproviderId": "2115950840",
				"categoryPaths": [
					"Serier > Killing Eve"
				],
				"categories": [
					"Killing Eve"
				],
				"cast": [
					"Oh,Sandra",
					"Shaw,Fiona",
					"Comer,Jodie",
					"Bodnia,Kim"
				],
				"blockedDevices": [],
				"audioLanguages": [],
				"assets": [
					{
						"subtitleLanguages": [],
						"streamingProtocol": "rtsp",
						"storeId": "8925",
						"resolution": "1080p",
						"protocol": "RTSP",
						"price": 0,
						"language": "swe",
						"is3d": false,
						"id": "18251879",
						"hastrailer": false,
						"format": "rtsp_FullHD",
						"devicetype": "STB",
						"codec": "mpeg4",
						"audioLanguages": [],
						"attributes": {
							"externalId": "ABXDN"
						}
					}
				],
				"age": 15,
				"titleId": "1963970510",
				"start": "2020-04-13T05:00:00Z",
				"description_long": "Slowly Slowly Catchy Monkey. Eve försöker bygga upp sitt liv igen efter att ha blivit skjuten av Villanelle. Villanelle försöker också gå vidare, tills hon kontaktas av en gammal fiende. Carolyn blir underminerad på jobbet och Kenny kan inte släppa sin egen utredning av The Twelve."
			}
		},
```

(Telia has content from HBO and other providers collected in one service). 

I think this looks promising for a sync or scrobbling service in your project, but I would need a more in depth guide for continued development on my part. 

Would it be possible to point me in the right direction or maybe even add Telia Play to the service list with info provided by me?

Thanks!

### PR DESCRIPTION
**Pull request by schteff**
A sync that fetches the watched info of as many episodes and movies as possible using the "Continue watching" list and "My list" features in Telia Play.

### CODE REVIEW

File: src/streaming-services/telia-play/telia-play.ts [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+			//Get watched status of all episodes / movies
+			const episodeIds = allEpisodes.map((e) => e.loopId);
+			const allIds = [...movieIds, ...episodeIds];
+			const watchedUrl = this.WATCHED_URL.replace('{EPISODE_IDS}', allIds.join());
+			const watchedResponse = await this.doGet(watchedUrl);
+			const watchedJson = JSON.parse(watchedResponse) as TeliaWatched;
+			const wMap = new Map(Object.entries(watchedJson.map));
+
+			//Limit 10% watched
+			const watchedEps = allEpisodes.filter((ep) => this.pctWatched(wMap.get(ep.loopId)) > 10);
```
Comment (MrMamen) [inline src/streaming-services/telia-play/telia-play.ts:237]: clever!

File: src/streaming-services/telia-play/telia-play.ts [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+	};
+
+	doGet = async (url: string) => {
+		const response = await Requests.send({
+			url: url,
+			method: 'GET',
+			headers: { Authorization: `Bearer ${this.jwt}` },
+		});
+
+		// console.log(url, JSON.parse(response));
```
Comment (MrMamen) [inline src/streaming-services/telia-play/telia-play.ts]: Remove this, I guess.

**Review by MrMamen — CHANGES_REQUESTED** [review]
Remember to update the readme file with support for teliaplay. You could also make a note somewhere that this only works with the Swedish version at this time.

**Comment by schteff** [comment]
Telia Play seems to only have a Swedish version. Even if I change language to English in the settings the website (https://www.teliaplay.se/) is in Swedish. I think the setting only changes the preferred audio language to English. 

So I just added a line to the list of supported streaming services in the readme.

**Review by MrMamen — DISMISSED** [review]
Sorry for the delay. See if you can find the error with matching movies on or around TraktSearch.ts line 152. Or you could just leave it as it is and file a bug while merging this.

**Comment by schteff** [comment]
Yeah, I added a fix for the mismatched movies. If only one search result is found, it is selected ignoring the title and year matching. 

I still experience many other weird "Not found" mismatches the first time I run the sync. But if I close the tab and run it again these disappear. If I manage to reproduce this I will file a bug or fix it later. 

I guess you can merge this pull request now :)

**Review by MrMamen — APPROVED** [review]

**Comment by MrMamen** [comment]
> Telia Play seems to only have a Swedish version. Even if I change language to English in the settings the website (https://www.teliaplay.se/) is in Swedish. I think the setting only changes the preferred audio language to English.
> 
> So I just added a line to the list of supported streaming services in the readme.

I found at least this:
https://www.teliaplay.lt
Seems like a localized clone to me.

**Comment by schteff** [comment]
Yeah, I can't sign in there with my account though. But I guess my sync code should work there to if the .lt domains are added to the whitelist.

**Comment by MrMamen** [comment]
I did something similar with viaplay. You don't need to do anything with it however. Can wait until someone requests it.

**Pull request merged**